src/query/atomWithInfiniteQuery.ts

```diff
@@ -1,45 +1,55 @@
-import { InfiniteQueryObserver, isCancelledError } from 'react-query'
+import { InfiniteQueryObserver, isCancelledError } from '@tanstack/query-core'
 import type {
   InfiniteData,
   InfiniteQueryObserverOptions,
   QueryKey,
   QueryObserverResult,
   RefetchOptions,
   RefetchQueryFilters,
-} from 'react-query'
+} from '@tanstack/query-core'
 import { atom } from 'jotai'
 import type { WritableAtom } from 'jotai'
 import { queryClientAtom } from './queryClientAtom'
 import { CreateQueryOptions, GetQueryClient } from './types'
 
-type AtomWithInfiniteQueryAction<TQueryFnData> =
-  | ({ type: 'refetch' } & Partial<
-      RefetchOptions & RefetchQueryFilters<TQueryFnData>
-    >)
+export type AtomWithInfiniteQueryAction<TQueryFnData> =
+  | {
+      type: 'refetch'
+      payload: Partial<RefetchOptions & RefetchQueryFilters<TQueryFnData>>
```

Comment:
This is a breaking change. Is it intentional?

Reply:
yep. we have discussed this a few years ago

Reply:
It's been a while... 😅